src/adapters/rubicon.js

```diff
@@ -188,57 +194,62 @@ var RubiconAdapter = function RubiconAdapter() {
   /**
    * Define the slot using the rubicontag.defineSlot API
    * @param {Object} Bidrequest
+   * @returns {RubiconSlot} Instance of RubiconSlot
    */
   function _defineSlot(bid) {
-    _rready(function () {
-      var newSlot = window.rubicontag.defineSlot({
-        siteId: bid.params.siteId,
-        zoneId: bid.params.zoneId,
-        id: bid.placementCode,
-        sizes: bid.params.sizes
-      });
-      if (bid.params.position) {
-        newSlot.setPosition(bid.params.position);
-      }
+    var id        = bid.placementCode;
+    var userId    = bid.params.userId;
+    var position  = bid.params.position;
+    var visitor   = bid.params.visitor || [];
+    var keywords  = bid.params.keywords || [];
+    var inventory = bid.params.inventory || [];
+    var slot      = window.rubicontag.defineSlot({
+      siteId : bid.params.siteId,
+      zoneId : bid.params.zoneId,
+      sizes  : bid.params.sizes,
+      id     : id
+    });
 
-      if (bid.params.userId) {
-        window.rubicontag.setUserKey(bid.params.userId);
-      }
+    if (userId) {
+      window.rubicontag.setUserKey(userId);
+    }
 
-      if (bid.params.keywords) {
-        for (var i = 0; i < bid.params.keywords.length; i++) {
-          newSlot.addKW(bid.params.keywords[i]);
-        }
-      }
+    if (position) {
+      slot.setPosition(position);
+    }
 
-      if (bid.params.inventory) {
-        for (var p in bid.params.inventory) {
-          if (bid.params.inventory.hasOwnProperty(p)) {
-            newSlot.addFPI(p, bid.params.inventory[p]);
-          }
-        }
-      }
+    for (var key in visitor) {
+      slot.setFPV(key, visitor[key]);
+    }
 
-      if (bid.params.visitor) {
-        for (var p in bid.params.visitor) {
-          if (bid.params.visitor.hasOwnProperty(p)) {
-            newSlot.addFPV(p, bid.params.visitor[p]);
-          }
-        }
-      }
-    });
+    for (var key in inventory) {
+      slot.setFPI(key, inventory[key]);
+    }
+
+    // For backwards compatibility check if function RubiconSlot#setKW exists.
+    if (slot.setKW) {
+      // reset keyword in order to handle pages re-using the same slot name
+      slot.setKW(keywords);
+
+      return slot;
+    }
+
+    slot.addKW(keywords);
+
+    return slot;
   }
 
   /**
    * Handle the bids received (from rubicon)
+   * @param {array} slots
    */
-  function _bidsReady() {
+  function _bidsReady(slots) {
     // NOTE: we don't really need to do anything,
     // because right now we're shimming XMLHttpRequest.open,
     // but in the future we'll get data from rubicontag here
     utils.logMessage('Rubicon Project bidding complete: ' + ((new Date).getTime() - _bidStart));
 
-    utils._each(rubicontag.getAllSlots(), function (slot) {
+    utils._each(slots, function (slot) {
```

Comment:
this is the point of the change.  We want to be able to refresh specific slots rather than being forced to do a full refresh each time.